hackathon_site/hardware/views.py

```diff
@@ -62,9 +60,32 @@ class IncidentListView(mixins.ListModelMixin, generics.GenericAPIView):
     filter_backends = (filters.DjangoFilterBackend, SearchFilter)
     filterset_class = IncidentFilter
 
+    def get_queryset(self):
+        queryset = Incident.objects.all()
+        if self.request.method == "GET":
+            return queryset.select_related(
+                "order_item", "order_item__order__team", "order_item__hardware"
+            )
+        elif self.request.method == "POST":
+            return queryset
+
+    def get_serializer_class(self):
+        if self.request.method == "GET":
+            return IncidentListSerializer
+        elif self.request.method == "POST":
+            return IncidentCreateSerializer
+
+    def get_permissions(self):
+        if self.request.method == "POST":
+            return [FullDjangoModelPermissions()]
```

Comment:
This should be the permissions class used no matter what.

As it is defined right now, anyone with an account could view all incidents.

Reply:
this was the way it was before I started. I'm assuming there had to be a reason for why the author of the GET feature didn't have permissions for GET requests.

Reply:
I think before we were sharing views between ppl with profiles and ppl with permissions but now we've switched to separate views for each so this view should have FullDjangoModelPermissions() for the GET as well since this is the view only for ppl with permissions

Reply:
Back when this view was originally implemented in #300, permissions were just left out to keep things simple. There's a comment saying they'll be added later [here](https://github.com/ieeeuoft/hackathon-template/pull/300#discussion_r715983131), looks like we never made a ticket for it though.

And as YuYing said, now it's pretty easy and we can just do `FullDjangoModelPermissions()` on all the admin-side views (you can just set `permission_classes = [FullDjangoModelPermissions]` on the view, no need for `get_permissions`).

Do you think you could really quickly add the basic checks for not logged in, logged in without permissions, and has permissions to `IncidentListsViewTestCase` (and maybe rename that to `IncidentListViewTestCase`? Should be a quick copy-paste.